pkg/loader/loader.go

```diff
@@ -142,9 +148,38 @@ func loadOpenAPI(prg *types.Program, data []byte) *openapi3.T {
 		prg.OpenAPICache = map[string]any{}
 	}
 
-	openAPIDocument, err = openapi3.NewLoader().LoadFromData(data)
-	if err != nil || openAPIDocument.Paths.Len() == 0 {
-		openAPIDocument = nil
+	if ver, ok := isOpenAPI(data); ok {
+		switch ver {
+		case 2:
+			// Convert OpenAPI v2 to v3
+			jsondata := data
```

Comment:
Not sure if this is intentional, but maybe this is more readable?
```suggestion
			var jsondata []byte
```